### ISSUE TITLE
Python program execution best practices (stop using os.system)

### ISSUE THREAD

**Issue by OnGle** [turnkeylinux/tracker#1221]
Turnkey uses `executil.system` quite a lot which uses `os.system` as a backend. `os.system` is generally considered bad practice because it launches a subshell and in certain circumstances can be susceptible to code injection.

This is mitigated somewhat in turnkey software by the use of `commands.mkarg` - an undocumented python function which should correctly escape command arguments so they don't cause issues. However the commands library was deprecated in python 2.6+ and removed entirely in python 3.

As such I propose we replace all mentions of `os.system` with that of an appropriate `subprocess` function.

**Comment by JedMeister** [comment]
Slightly edited your OP @OnGle, hope you don't mind. Plus I added the "core" tag. Whilst it doesn't apply only to core, I think that it's worth adding something appliance/project related. Also pinned it to the v16.0 milestone. Whether or not we do that for v16.0 still remains to be seen, but let's at least pin it there for now and see.

Also, perhaps a little off topic, but I recently came across [grumpy](https://github.com/grumpyhome/grumpy) (a python -> go source code transcompiler and runtime) which might be worth consideration?! If nothing else, IMO moving inithooks to go for v16.0 makes a ton of sense... Interested in discussing further.
